src/System.Net.Sockets/src/System/Net/Sockets/UDPClient.cs

```diff
@@ -424,7 +440,7 @@ public void JoinMulticastGroup(IPAddress multicastAddr)
 
             // IPv6 Changes: we need to create the correct MulticastOption and
             //               must also check for address family compatibility.
-            if (multicastAddr.AddressFamily != _family)
+            if (!IsAddressFamilyCompatible(multicastAddr.AddressFamily))
```

Comment:
You should verify that this actually works on DualMode socket. 
We have seen various anomalies when dealing with multicast.

Reply:
I've added a test for that, however, the one on Linux timed out so maybe needs to be re-run. But the one on [NETFX](https://mc.dot.net/#/user/nbaztec/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/3d0a10911df979bdee180a49a13e682f54edfbe2/workItem/System.Net.Sockets.Tests/analysis/xunit/System.Net.Sockets.Tests.UdpClientTest~2FBeginSend_IPv6Socket_IPv4Dns_Success), I'm not sure what the problem is about, can use some input there.